lib/Mailer.php

```diff
@@ -32,46 +34,60 @@ abstract class Mailer implements MailerInterface
     private $messageFactory;
 
     /**
-     * @param HttpClient $httpClient
-     * @param string     $serverToken
+     * @var RequestFactoryInterface|null
      */
-    public function __construct(HttpClient $httpClient, $serverToken)
+    private $requestFactory;
+
+    /**
+     * @var StreamFactoryInterface|null
+     */
+    private $streamFactory;
+
+    public function __construct(ClientInterface $httpClient, string $serverToken)
     {
         $this->setHttpClient($httpClient);
         $this->setServerToken($serverToken);
     }
 
-    /**
-     * @param HttpClient $httpClient
-     */
-    public function setHttpClient(HttpClient $httpClient)
+    public function setHttpClient(ClientInterface $httpClient)
```

Comment:
This is technically a BC break, right?

Reply:
Well, only if a child class override the method, which I consider OK as none of our core mailer has a use case for that.

For callers, this is OK, as `ClientInterface` is a parent type of the previous `HttpClient` (HTTPlug 2+ implements PSR-18)